### ISSUE TITLE
Problems with container that runs forever

### ISSUE THREAD

**Issue by Smeds** [ansible-community/ansible-bender#156]
https://github.com/ansible-community/ansible-bender/blob/368e39e776f938efe4b723bbedbb83ecbd07a299/ansible_bender/builders/buildah_builder.py#L341

How should one handle docker containers that are suppose to run as service, i.e they will not stop unless one manually stop the container. For example [stackstorm](https://github.com/StackStorm/st2-docker/blob/master/images/stackstorm/bin/entrypoint.sh) docker image, right now ansible-bender build will get stuck after podman have tried to run the image, since the image never stops.

**Comment by Smeds** [comment]
The stackstorm entrypoint file, that I link to, will keep the container running indefinitely

**Comment by Smeds** [comment]
I have been able to build by adding  r"--entrypoint=""" to the podman commands. But I would prefer to have it as an input flag, and option to override entrypoint value during build.

**Comment by TomasTomecek** [comment]
Hey, thanks for opening this issue!

Pretty good suggestion to ale make entrypoint configurable.

**Comment by TomasTomecek** [comment]
I'd add a new option to the `build` command to configure the entrypoint:
```
$ ab build --entrypoint='/usr/bin/lol' ...
```

And also add it to the metadata:
```yaml
  vars:
    ansible_bender:
      working_container:
        entrypoint: /usr/bin/lol
```

Does that make sense?

**Comment by kmehant** [comment]
@TomasTomecek I suppose this issue is fixed now, shall we close this?

**Comment by TomasTomecek** [comment]
Oh yeah, good point!

**Comment by seidzade** [comment]
Hi guys, 

Seems like the issue doesn't fixed in latest release , I have still unable to run build with entrypoint defined under "working_container" - property are not allowed , the property even not in schema.py :

`There was an error during execution: Additional properties are not allowed ('entrypoint' was unexpected)

Failed validating 'additionalProperties' in schema['properties']['working_container']:
    {'$id': '#/properties/working_container',
     'additionalProperties': False,
     'properties': {'user': {'$id': '#/properties/working_container/properties/user',
                             'examples': ['leonardo'],
                             'title': 'UID or username used to invoke the '
                                      'container',
                             'type': ['string', 'null']},
                    'volumes': {'$id': '#/properties/working_container/properties/volumes',
                                'items': {'$id': '#/properties/working_container/properties/volumes/items',
                                          'examples': ['/path/to/a/directory'],
                                          'title': 'The Items Schema',
                                          'type': 'string'},
                                'title': 'A list of paths which are meant '
                                         'to be hosted outside of the '
                                         'container',
                                'type': 'array'}},
     'title': 'settings for the container where the build occurs',
     'type': ['object', 'null']}

On instance['working_container']:`

Also --entrypoint flag in build command not working , podman still run without overwriting the image defined entrypoint :

`09:58:47.711 utils.py          INFO   running command: "['podman', 'run', '--rm', 'artifactory.rdlab.local/docker-virtual/company/centos_supervisord', 'true']"`  


My setup :

`Name: ansible-bender
Version: 0.7.1.dev89+ga04a4a2
Summary: A tool which builds container images using Ansible playbooks
Home-page: https://github.com/TomasTomecek/ansible-bender
Author: Tomas Tomecek
Author-email: tomas@tomecek.net
License: MIT
Location: /usr/local/lib/python3.6/site-packages
Requires: PyYAML, tabulate, jsonschema`

**Comment by TomasTomecek** [comment]
@seidzade you seem to be using a version which doesn't have entrypoints implemented yet, please make sure to run ansible-bender 0.8.0

```
$ ansible-bender -V
0.8.0
```

I just verified it works correctly.

Making sure our playbook has the entrypoint set up correctly:
```
$ grep entry full_conf_pb.yaml
...
  vars:
    ansible_bender:
      target_image:
        entrypoint: great-entry-point
...
```
```
$ ansible-bender build full_conf_pb.yaml

PLAY [funky-mona-lisa-20191120-161017283723-cont] ***************************************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************************************************************
ok: [funky-mona-lisa-20191120-161017283723-cont]

TASK [command] **************************************************************************************************************************************************************
changed: [funky-mona-lisa-20191120-161017283723-cont]

PLAY RECAP ******************************************************************************************************************************************************************
funky-mona-lisa-20191120-161017283723-cont : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

Getting image source signatures
Copying blob sha256:20903165877b120f7b705ed38f1f2d0fde6986fbebea4033781625db4f0a2601
Copying blob sha256:e6842940d106906c2351f0d94f30a9db4944067e53687010655e257a5f766975
Copying config sha256:79d84bd6946353b17857282ab177c655f5b69799ea6591afd429de0e1def546f
Writing manifest to image destination
Storing signatures
79d84bd6946353b17857282ab177c655f5b69799ea6591afd429de0e1def546f
Image 'funky-mona-lisa' was built successfully \o/
```

Now veryifing the image metadata:
```
$ buildah inspect funky-mona-lisa | grep -i entry
            "Entrypoint": [
                "great-entry-point"

$ podman inspect funky-mona-lisa | grep -i entry
            "Entrypoint": [
                "great-entry-point"
```

CLI now:
```
$ ansible-bender build --entrypoint /asdqwe full_conf_pb.yaml

PLAY [funky-mona-lisa-20191120-161530988871-cont] ***************************************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************************************************************
ok: [funky-mona-lisa-20191120-161530988871-cont]

TASK [command] **************************************************************************************************************************************************************
changed: [funky-mona-lisa-20191120-161530988871-cont]

PLAY RECAP ******************************************************************************************************************************************************************
funky-mona-lisa-20191120-161530988871-cont : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

Getting image source signatures
Copying blob sha256:20903165877b120f7b705ed38f1f2d0fde6986fbebea4033781625db4f0a2601
Copying blob sha256:9b81b3645025e54b6d2a526249faf6863a2e83ee88ec87851a6aa7accb65b5ce
Copying config sha256:dcff2f77d11d7d7c8b75e83188a140e07d59646e0db8fa054f67f89ce36ea35d
Writing manifest to image destination
Storing signatures
dcff2f77d11d7d7c8b75e83188a140e07d59646e0db8fa054f67f89ce36ea35d
Image 'funky-mona-lisa' was built successfully \o/

$ podman inspect funky-mona-lisa | grep -C 3 -i entry
                "FBR=f31-updates-candidate",
                "z=value"
            ],
            "Entrypoint": [
                "/bin/sh",
                "-c",
                "/asdqwe"
```

**Comment by seidzade** [comment]
@TomasTomecek tnx to reply :) , I already upgraded to 0.8.0 released today.

But, correct me if I wrong, here you guys discussed about "entrypoint" in "working_container" context, not "target_image" . My understanding is that the problem happens during build-time container creation by podman, but "target_image" --> "entrypoint" affects entrypoint of the "production" container and not related to image build-time.  
@TomasTomecek , attaching your example from one of the above messages:
`  vars:
    ansible_bender:
      **working_container**:
        entrypoint: /usr/bin/lol`

BTW, maybe another approach can be suggested, for example, pass "-d" flag to podman , in order to release the command in cases where base image already has "entrypoint". But I'm not sure if such approach is acceptable. Better to overwrite the existing entrypoint during buil-time.

**Comment by TomasTomecek** [comment]
ah okay, so you want to change the entrypoint of the working container - that's something different :)

could you please create a new issue for that?

**Comment by seidzade** [comment]
> ah okay, so you want to change the entrypoint of the working container - that's something different :)
> 
> could you please create a new issue for that?

yes , sure , will do it this evening

**Comment by Smeds** [comment]
Sorry for not responding! I have been away on paternity leave. 

Issue #198 is exactly what I am looking for.
